src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/BindingSourceDesigner.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel.Design;
+using System.ComponentModel;
+
+namespace System.Windows.Forms.Design;
+
+internal class BindingSourceDesigner : ComponentDesigner
+{
+    private bool _bindingUpdatedByUser;
+
+    /// <summary>
+    ///  Flag set by UITypeEditors to notify us when the BindingSource binding has been
+    ///  manually changed by the user (ie. DataSource or DataMember value was edited).
+    ///
+    ///  In response to this action, we will send a notification to the DataSourceProviderService
+    ///  once the corresponding ComponentChanged event comes through (indicating that the new
+    ///  binding has actually taken effect).
+    ///
+    ///  This notification will cause the DataSourceProviderService to re-examine the data source
+    ///  that the BindingSource is now bound to, so it can generate any table adapters or fill
+    ///  statements necessary to properly set up that data source.
+    ///
+    ///  This serves the scenario of an advanced user configuring a BindingSource manually. In
+    ///  normal user scenarios, BindingSources get auto-generated as a result of various drag/drop
+    ///  operations, and in those scenarios the above notification is sent automatically.
+    ///
+    ///  VSWhidbey#256272
+    /// </summary>
+    public bool BindingUpdatedByUser
+    {
+        set
+        {
+            _bindingUpdatedByUser = value;
+        }
+    }
+
+    /// <summary>
+    ///  Initialize this designer.
+    /// </summary>
+    public override void Initialize(IComponent component)
+    {
+        base.Initialize(component);
+
+        IComponentChangeService componentChangeSvc = (IComponentChangeService)GetService(typeof(IComponentChangeService));
+        if (componentChangeSvc is not null)
+        {
+            componentChangeSvc.ComponentChanged += new ComponentChangedEventHandler(OnComponentChanged);
+            componentChangeSvc.ComponentRemoving += new ComponentEventHandler(OnComponentRemoving);
```

Comment:
Please remove unnecessary new for event handlers in other places in this PR as well
```suggestion
            componentChangeSvc.ComponentChanged += OnComponentChanged;
            componentChangeSvc.ComponentRemoving += OnComponentRemoving;
```